Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Leads.hs

```diff
@@ -0,0 +1,81 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.Leads
+  ( LeadsReq (..),
+    createLeadsDetails,
+  )
+where
+
+import qualified Domain.Types.Leads as DLeads
+import qualified Domain.Types.Merchant as DMerchant
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto (runTransaction)
+import qualified Kernel.Storage.Esqueleto as Esq
+import qualified Kernel.Types.APISuccess as APISuccess
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Types.SlidingWindowLimiter (APIRateLimitOptions)
+import Kernel.Types.Version
+import Kernel.Utils.Common
+import Kernel.Utils.SlidingWindowLimiter
+import Storage.CachedQueries.CacheConfig
+import qualified Storage.CachedQueries.Merchant as QMerchant
+import qualified Storage.Queries.Leads as QLeads
+
+data LeadsReq = LeadsReq
+  { deviceToken :: Text,
+    source :: Maybe Text,
+    merchantId :: ShortId DMerchant.Merchant
+  }
+  deriving (Generic, ToJSON, FromJSON, ToSchema)
+
+leadsHitsCountKey :: Text -> Text
+leadsHitsCountKey deviceToken = "BAP:Leads:createLeadsDetails:" <> deviceToken <> ":hitsCount"
+
+createLeadsDetails ::
+  ( HasFlowEnv m r '["apiRateLimitOptions" ::: APIRateLimitOptions],
+    EsqDBFlow m r,
+    EncFlow m r,
+    CacheFlow m r
+  ) =>
+  LeadsReq ->
+  Maybe Version ->
+  Maybe Version ->
+  Maybe Text ->
+  m APISuccess.APISuccess
+createLeadsDetails req mbBundleVersion mbClientVersion mbPlatform = do
+  checkSlidingWindowLimit (leadsHitsCountKey req.deviceToken)
+  merchant <- QMerchant.findByShortId req.merchantId >>= fromMaybeM (MerchantDoesNotExist req.merchantId.getShortId)
+  mbLeads <- QLeads.findyByLeadsAndSource req.deviceToken req.source merchant.id
+  when (isJust mbLeads) $ do
+    Esq.runTransaction $ QLeads.updateLeads req.deviceToken req.source merchant.id
```

Comment:
if it is already create, why to update again?

Reply:
It is only to update the timestamp in updatedAt and shows the error